### PR TITLE
11016 icon color lost

### PR DESCRIPTION
**Pull request by ivanmalagon**
Related to #11016

The issue was that the change in color via picker triggers a `render`call. The template has an `IMG` tag that is immune to the `fill` CSS property. We have to reload the `IMG` element with the proper `SVG`.

⚠️  There's an issue with CORS that provoke a fail in the request for the SVG. @javierarce can work with no problem in his Chrome but if you have to test this issue it's likely that you can't.

### CODE REVIEW

**Comment by javierarce** [comment]
I've reviewed the code of the PR and found that the current solution triggers a request every time the color picker is dragged.

requests

Talking with @ivanmalagon, we've decided that a good way to solve this problem would be caching the requests so we don't need to ask for the same SVG file every time.

**Comment by ivanmalagon** [comment]
SVG _cached_ via property to avoid multiple requests while picking color. cc @javierarce

**Comment by matallo** [comment]
LGTM, merged with master and prepared for acceptance

**Comment by ivanmalagon** [comment]
One test is not passing in Travis. Probably because of differences between Phantom in Travis.

Fixing it.

**Comment by gfiorav** [comment]
Hi!

I'm missing some acceptance instruction. I know I need `icon-styling` feature flag. Anything else? Something that should be checked apart from the main purpose?

File: .gitignore [hunk 1]
```diff
@@ -70,3 +70,4 @@ doc/manual/build
 .byebug_history
 /tags
 /.tags_swap
+.vscode/
```
Comment (xavijam) [inline .gitignore]: ?

File: app/assets/stylesheets/editor-3/assets.css.scss [hunk 2]
```diff
@@ -29,11 +29,7 @@
 }
 
 .AssetItem-icon {
-  width: 100%;
-  height: 100%;
-  background-repeat: no-repeat;
-  background-position: center;
-  background-size: 24px 24px;
+  vertical-align: middle;
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/assets.css.scss]: And if it is bigger?

File: app/assets/stylesheets/editor-3/tab-pane.css.scss [hunk 3]
```diff
@@ -8,8 +8,12 @@
   min-height: 0;
 }
 
-.Tab-paneLabel-image {
+.Tab-paneLabel-image-container {
   display: inline-block;
-  max-width: 15px;
-  height: 15px;
+  min-width: 18px;
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/tab-pane.css.scss]: And max width?

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-file-view.js [hunk 4]
```diff
@@ -93,7 +93,7 @@ module.exports = CoreView.extend({
         url: icon.get('public_url'),
         action: action,
         template: function () {
-          return AssetTemplate({ type: type, public_url: icon.get('public_url'), name: icon.get('name') });
+          return AssetTemplate({ type: type, public_url: icon.get('public_url') + '?req=markup', name: icon.get('name') });
```
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-file-view.js]: What does that req param mean?

File: lib/assets/core/javascripts/cartodb3/components/img-loader-view.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+var $ = require('jquery');
+var CoreView = require('backbone/core-view');
+
+function endsWith (str, suffix) {
```
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/img-loader-view.js]: Could we move this function to js-utils helper class?

**Comment by xavijam** [comment]
Hey, there is a problem with the CSS after starting acceptance.

screen shot 2016-12-28 at 16 24 39

**Comment by xavijam** [comment]
Also fill component is opened too slow when an image/icon is selected, are you suffering the same thing?

File: .gitignore (hunk 1 above)
Comment (ivanmalagon) [inline .gitignore]: That folder is local to every team member that is using VS Code as editor. It contains local settings for the project related to the editor and it must be ignored.

Now I'm the only one that is using VS Code, but as the time pass, everyone will be converted.

File: app/assets/stylesheets/editor-3/tab-pane.css.scss (hunk 3 above)
Comment (ivanmalagon) [inline app/assets/stylesheets/editor-3/tab-pane.css.scss]: It must be topped by height, not width. Checked with @piensaenpixel when selecting long icons.

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-file-view.js (hunk 4 above)
Comment (ivanmalagon) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-file-view.js]: Just to differentiate requests to resources coming from the DOM or Ajax. We had a problem with browser caching the resource if present in the markup and then asking for the very same resource via Ajax.

After differentiating the resource via query string this problem vanished. I don't know why is needed also in the markup (and not only Ajax) but empirically this works.

File: app/assets/stylesheets/editor-3/tab-pane.css.scss (hunk 3 above)
Comment (xavijam) [inline app/assets/stylesheets/editor-3/tab-pane.css.scss]: :+1:

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-file-view.js (hunk 4 above)
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-file-view.js]: Aham!

File: app/assets/stylesheets/editor-3/assets.css.scss (hunk 2 above)
Comment (ivanmalagon) [inline app/assets/stylesheets/editor-3/assets.css.scss]: Big icons now fit properly.

File: lib/assets/core/javascripts/cartodb3/components/img-loader-view.js (hunk 5 above)
Comment (ivanmalagon) [inline lib/assets/core/javascripts/cartodb3/components/img-loader-view.js]: Done!

**Comment by ivanmalagon** [comment]
Problem with row being taller than expected fixed. https://github.com/CartoDB/cartodb/pull/11067#issuecomment-269492191

File: .gitignore (hunk 1 above)
Comment (xavijam) [inline .gitignore]: ha! ok but you are trying to convince us!

File: app/assets/stylesheets/editor-3/assets.css.scss (hunk 2 above)
Comment (xavijam) [inline app/assets/stylesheets/editor-3/assets.css.scss]: :+1:

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/assets-list-view.js [hunk 6]
```diff
@@ -65,10 +65,10 @@ module.exports = CoreView.extend({
 
   _selectItem: function (m) {
     this.model.set('image', m.get('public_url'));
-    this._unselectItems(m);
+    this.unselectItems(m);
   },
 
-  _unselectItems: function (m) {
+  unselectItems: function (m) {
```
Comment (xavijam) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/fill/input-color/assets-list-view.js]: "public"?

**Comment by xavijam** [comment]
Passing acceptance!

**Pull request merged**